servers/jax-rs-tests/pom.xml

```diff
@@ -124,6 +130,11 @@
       <artifactId>logback-classic</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.iceberg</groupId>
+      <artifactId>iceberg-spark3-runtime</artifactId>
```

Comment:
Totally agree. But now it is not failing because there is conflict. But it can fail once we upgrade the versions.
As it a whole testcase refactor, I have to figure out / adjust this testcases. **So, will do it in an immediate follow up** if it is ok. 

 #3601 - Created this ticket to track it. 

I am planning to move these tests to the module (`nessie-spark-extensions-base`) that already depends on
`iceberg-spark3-runtime`. But it will become IT test. which should be fine as it is anyway end to end test.

Reply:
Also thought of adding individual jars and exclude nessie-client from iceberg. But once Iceberg upgrades these testcases will fail in that case. So, I felt moving to `nessie-spark-extensions-base` is the right way.

Reply:
added exclusions to the iceberg's nessie classes and depending on the latest nessie version. 

with a follow up of https://github.com/projectnessie/nessie/issues/3692 once Iceberg makes a release.